src/skalo/positioning.rs

```diff
@@ -0,0 +1,256 @@
+use hashbrown::{HashMap, HashSet};
+use seq_io::fasta::{Reader, Record};
+use std::cmp::Ordering;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+use flate2::read::MultiGzDecoder;
+
+use crate::skalo::utils::{encode_kmer, rev_compl, VariantInfo};
+
+// extract genomic k-mers with up to 3 distinct positions
+pub fn extract_genomic_kmers(
+    file_path: PathBuf,
+    k: usize,
+) -> (HashMap<u128, Vec<u32>>, Vec<u8>, String) {
```

Comment:
Return type complex – I think some thought about having a skalo struct to simplify the overall flow of the process would help a lot